packit/local_project.py

```diff
@@ -67,6 +75,7 @@ def __init__(
         merge_pr: bool = True,
         target_branch: str = "",
         working_dir_temporary: bool = False,
+        specfile: Specfile = None,
```

Comment:
It looks like the specfile handling is really tricky. I would try to avoid this TBH. To me, `LocalProject` is just about git project (either local or remote) and has no relation to specfile. We've been thinking already about moving this class away from Packit and this change would make the step backwards.

I understand this is really handy and won't be easy to solve in another way but I still think we should follow the original purpose of this class and don't put any unrelated code into it.

Reply:
I agree, I didn't like the idea to pass the specfile to this class either.
But I don't think the `LocalProject` should be just a git project. We could take the project from somewhere else than git, like from a Specfile. Probably the `LocalProject` builder is the right place where to pass a specfile for building the `LocalProject` in a new way, but I think we are not yet using it 🤔, right?

Reply:
I was thinking to make the `LocalProject` having a `packit.repo.NonRepo` object; an object with the same API of `Repo` but with some "empty" implementations like clone or checkout, that just print a warning and do nothing.
In this way I can probably get rid of the `up_is_git_repo` but I wanted to discuss it before taking this way.

If we don't want to have a `repo.NonRepo` object  that should be probably injected into the `LocalProject` by the `LocalProjectBuilder` then I think we need to refactoring the actions code. Now the actions are a feature of the `PackitRepositoryBase` object  which is supposed working with a `repo` object inside `LocalProject`. But  for being able to run an action on a project I don't really need a git repo. So the other clean way I see to implement this is moving away the command_handler from the `PackitRepositoryBase`, and make it work even with a non git repo.
And this is another topic I would like to write down better and discuss.  I don't know what you could think about this?